maskit/masks.py

```diff
@@ -0,0 +1,296 @@
+        return mask
+
+    @property
+    def layer_mask(self):
+        """Returns the encapsulated layer mask."""
+        return self._layers.mask
+
+    @property
+    def wire_mask(self):
+        """Returns the encapsulated wire mask."""
+        return self._wires.mask
+
+    @property
+    def parameter_mask(self):
+        """Returns the encapsulated parameter mask."""
+        return self._parameters.mask
+
+    def perturb(
+        self,
+        axis: PerturbationAxis,
+        amount: Optional[int] = None,
+        mode: PerturbationMode = PerturbationMode.INVERT,
+    ):
+        """
+        Perturbs the MaskedCircuit for a given ``axis`` that is of type
+        :py:class:`~.PerturbationAxis`. The perturbation is applied ``amount``times
+        and depends on the given ``mode`` of type :py:class:`~.PerturbationMode`.
+        If no amount is given, that is ``amount=None``, a random ``amount`` is
+        determined given by the actual size of the py:attr:`~.mask`. The ``amount``
+        is automatically limited to the actual size of the py:attr:`~.mask`.
+
+        :param amount: Number of items to perturb, defaults to None
+        :param axis: Which mask to perturb
+        :param mode: How to perturb, defaults to PerturbationMode.INVERT
+        :raises AssertionError: Raised for negative amounts
+        :raises NotImplementedError: Raised in case of an unknown mode
+        """
+        assert (
+            amount is None or amount >= 0
+        ), "Negative values are not supported, please use PerturbationMode.REMOVE"
+        assert mode in list(PerturbationMode), "The selected mode is not supported."
+        if amount == 0:
+            return
+        if axis == PerturbationAxis.LAYERS:
+            mask_fn = self._mask_layers
+        elif axis == PerturbationAxis.WIRES:
+            mask_fn = self._mask_wires
+        elif axis == PerturbationAxis.RANDOM:  # Axis is on parameters
+            mask_fn = self._mask_parameters
+        else:
+            raise NotImplementedError(f"The perturbation {axis} is not supported")
+        mask_fn(amount=amount, mode=mode)
+
+    def _mask_layers(self, amount, mode):
+        self._layers.perturb(amount=amount, mode=mode)
+
+    def _mask_wires(self, amount, mode):
+        self._wires.perturb(amount=amount, mode=mode)
+
+    def _mask_parameters(self, amount, mode):
+        self._parameters.perturb(amount=amount, mode=mode)
+
+    def reset(self):
+        """Resets all masks."""
+        self._layers.reset()
+        self._wires.reset()
+        self._parameters.reset()
+
+    def apply_mask(self, values: np.ndarray):
+        """
+        Applies the encapsulated py:attr:`~.mask`s to the given ``values``.
+        Note that the values should have the same shape as the py:attr:`~.mask`.
+
+        :param values: Values where the mask should be applied to
+        :raises AssertionError: In case the shape of values and mask don't match.
+        """
```

Comment:
Minor point, but the description does not make it clear to me what the method actually does. I assume ``values`` is an array of booleans, and every truthy position gets masked?